docs/sdk/src/reference_docs/omni_node.rs

```diff
@@ -0,0 +1,184 @@
+//!
+//! ## Appendix
+//!
+//! This section describes how the interdependence between the node and the runtime is related to
+//! the consensus engine. This information is useful for those who want to understand the
+//! historical context of the node and the runtime.
+//!
+//! ### Consensus Engine
+//!
+//! In any given substrate-based chain, both the node and the runtime will have their own
+//! opinion/information about what consensus engine is going to be used.
+//!
+//! In practice, the majority of the implementation of any consensus engine is in the node side, but
+//! the runtime also typically needs to expose a custom runtime-api to enable the particular
+//! consensus engine to work, and that particular runtime-api is implemented by a pallet
+//! corresponding to that consensus engine.
+//!
+//! For example, taking a snippet from [`solochain_template_runtime`], the runtime has to provide
+//! this additional runtime-api (compared to [`minimal_template_runtime`]), if the node software is
+//! configured to use the Aura consensus engine:
+//!
+//! ```text
+//! impl sp_consensus_aura::AuraApi<Block, AuraId> for Runtime {
+//!     fn slot_duration() -> sp_consensus_aura::SlotDuration {
+//!         ...
+//!     }
+//!     fn authorities() -> Vec<AuraId> {
+//!         ...
+//!     }
+//! }
+//! ```
+//!
+//! For simplicity, we can break down "consensus" into two main parts:
+//!
+//! * Block Authoring: Deciding who gets to produce the next block.
+//! * Finality: Deciding when a block is considered final.
+//!
+//! For block authoring, there are a number of options:
+//!
+//! * [`sc_consensus_manual_seal`]: Useful for testing, where any node can produce a block at any
+//!   time. This is often combined with a fixed interval at which a block is produced.
+//! * [`sc_consensus_aura`]/[`pallet_aura`]: A simple round-robin block authoring mechanism.
+//! * [`sc_consensus_babe`]/[`pallet_babe`]: A more advanced block authoring mechanism, capable of
+//!   anonymizing the next block author.
+//! * [`sc_consensus_pow`]: Proof of Work block authoring.
+//!
+//! For finality, there is one main option shipped with polkadot-sdk:
+//!
+//! * [`sc_consensus_grandpa`]/[`pallet_grandpa`]: A finality gadget that uses a voting mechanism to
+//!   decide when a block
+//!
+//! **The most important lesson here is that the node and the runtime must have matching consensus
+//! components.**
+//!
+//! ### Consequences for OmniNode
+//!
+//!
+//! The consequence of the above is that anyone using the OmniNode must also be aware of the
+//! consensus system used in the runtime, and be aware if it is matching that of the OmniNode or
+//! not. For the time being, [`polkadot-omni-node`] only supports:
+//!
+//! * Parachain-based Aura consensus, with 6s async-backing block-time, and before full elastic
+//!   scaling). [`polkadot_omni_node_lib::cli::Cli::experimental_use_slot_based`] for fixed factor
+//!   scaling (a step
```

Comment:
This sentence is a bit mangled. IMO we should not mention elastic scaling here, because I expect it to be advanced usage. Most chains will probably never require it.

Reply:
Fair enough